cmd/xl-v1-healing.go

```diff
@@ -232,7 +233,7 @@ func healObject(ctx context.Context, storageDisks []StorageAPI, bucket string, o
 	latestDisks, modTime := listOnlineDisks(storageDisks, partsMetadata, errs)
 
 	// List of disks having all parts as per latest xl.json.
-	availableDisks, dataErrs := disksWithAllParts(ctx, latestDisks, partsMetadata, errs, bucket, object)
+	availableDisks, dataErrs := disksWithAllParts(ctx, latestDisks, partsMetadata, errs, bucket, object, scanMode == madmin.HealDeepScan)
```

Comment:
you could pass `scanMode` to disksWithAllParts as is, and in disksWithAllParts() you can do
`if scanMode == madmin.HealDeepScan {}` it's more scalable/readable, i.e in case we add additional modes that need to be handled by disksWithAllParts()

Reply:
> you could pass `scanMode` to disksWithAllParts as is, and in disksWithAllParts() you can do
> `if scanMode == madmin.HealDeepScan {}` it's more scalable/readable, i.e in case we add additional modes that need to be handled by disksWithAllParts()

Done